bson/src/main/scala/org/mongodb/scala/bson/codecs/IterableCodec.scala

```diff
@@ -0,0 +1,123 @@
+    new IterableCodec(registry, bsonTypeClassMap, valueTransformer.getOrElse(DEFAULT_TRANSFORMER))
+  }
+
+  private val DEFAULT_TRANSFORMER = new Transformer() {
+    def transform(objectToTransform: Object): Object = objectToTransform
+  }
+}
+
+/**
+ * Encodes and decodes `Iterable` objects.
+ *
+ * @since 1.2
+ */
+@SuppressWarnings(Array("rawtypes"))
+case class IterableCodec(registry: CodecRegistry, bsonTypeClassMap: BsonTypeClassMap, valueTransformer: Transformer) extends Codec[Iterable[_ <: Any]] {
+  lazy val bsonTypeCodecMap = new BsonTypeCodecMap(bsonTypeClassMap, registry)
+
+  override def decode(reader: BsonReader, decoderContext: DecoderContext): Iterable[_] =
+    readValue(reader, decoderContext).asInstanceOf[Iterable[_]]
+
+  override def encode(writer: BsonWriter, value: Iterable[_ <: Any], encoderContext: EncoderContext): Unit =
+    writeValue(writer, encoderContext, value)
+
+  override def getEncoderClass: Class[Iterable[_]] = classOf[Iterable[_]]
+
+  @SuppressWarnings(Array("unchecked", "rawtypes"))
+  private def writeValue[T](writer: BsonWriter, encoderContext: EncoderContext, value: T) {
+    value match {
+      case isNull if value == null => writer.writeNull() // scalastyle:ignore
+      case map: Map[_, _] =>
+        writeMap(writer, map.asInstanceOf[Map[String, Any]], encoderContext.getChildContext)
+      case list: Iterable[_] =>
+        writeIterable(writer, list, encoderContext.getChildContext)
+      case _ =>
+        val codec = registry.get(value.getClass).asInstanceOf[Encoder[T]]
+        encoderContext.encodeWithChildContext(codec, writer, value)
+    }
+  }
+
+  private def writeMap(writer: BsonWriter, map: Map[String, Any], encoderContext: EncoderContext) {
+    writer.writeStartDocument()
+    map.foreach(kv => {
+      writer.writeName(kv._1)
+      writeValue(writer, encoderContext, kv._2)
+    })
+    writer.writeEndDocument()
+  }
+
+  private def writeIterable(writer: BsonWriter, list: Iterable[_], encoderContext: EncoderContext) {
+    writer.writeStartArray()
+    list.foreach(value => writeValue(writer, encoderContext, value))
+    writer.writeEndArray()
+  }
+
+  private def readValue(reader: BsonReader, decoderContext: DecoderContext): Any = {
+    reader.getCurrentBsonType match {
+      case BsonType.NULL =>
+        reader.readNull()
+        null // scalastyle:ignore
+      case BsonType.ARRAY => readList(reader, decoderContext)
+      case BsonType.DOCUMENT => readMap(reader, decoderContext)
+      case BsonType.BINARY if BsonBinarySubType.isUuid(reader.peekBinarySubType) && reader.peekBinarySize == 16 =>
+        registry.get(classOf[UUID]).decode(reader, decoderContext)
+      case bsonType: BsonType => valueTransformer.transform(bsonTypeCodecMap.get(bsonType).decode(reader, decoderContext))
+    }
+  }
+
+  private def readMap(reader: BsonReader, decoderContext: DecoderContext): Map[String, _] = {
```

Comment:
The other reason is to make encoding and decoding round trip.  

If I delegated to the registry then `List(Map("a" -> 1, "b" -> 2, "c" -> null))` would come back as: `List(BsonDocument(...))`.   BsonDocument is not a scala Map so that fails.